extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/RestClientBuilderHelper.java

```diff
@@ -112,4 +110,58 @@ public static Sniffer createSniffer(RestClient client, ElasticsearchConfig confi
 
         return builder.build();
     }
+
+    public enum EsAuth {
+        NONE {
+            @Override
+            public void apply(HttpAsyncClientBuilder httpClientBuilder, ElasticsearchConfig config) {
+                // No authentication needed
+            }
+        },
+
+        BASIC {
+            @Override
+            public void apply(HttpAsyncClientBuilder httpClientBuilder, ElasticsearchConfig config) {
+                if (config.username().isPresent()) {
+                    if (!"https".equalsIgnoreCase(config.protocol())) {
+                        LOG.warn("Using Basic authentication in HTTP implies sending plain text passwords over the wire, "
+                                + "use the HTTPS protocol instead.");
+                    }
+
+                    CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+                    credentialsProvider.setCredentials(AuthScope.ANY,
+                            new UsernamePasswordCredentials(config.username().get(),
+                                    config.password().orElse(null)));
+                    httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+                }
+            }
+        },
+
+        API_KEY {
+            @Override
+            public void apply(HttpAsyncClientBuilder httpClientBuilder, ElasticsearchConfig config) {
+                if (config.esApiKeyAuth().isPresent()) {
+                    ElasticsearchConfig.EsApiKeyAuth auth = config.esApiKeyAuth().get();
+                    Header apiKeyHeader = new BasicHeader(HttpHeaders.AUTHORIZATION,
+                            "ApiKey " + auth.apiKey());
+                    httpClientBuilder.setDefaultHeaders(Collections.singleton(apiKeyHeader));
+                    LOG.info("API Key authentication is enabled.");
```

Comment:
```suggestion
```
I'd say let's be consistent here, since we aren't logging at info for the basic auth, we shouldn't log here too.